package/tests/test_server.py

```diff
@@ -243,6 +261,128 @@ def test_nodes_endpoint(cli_runner, client):
     assert data == EXPECTED_PIPELINE_DATA
 
 
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_pipeline_flag(cli_runner, client):
+    """Test that running viz with `--pipeline` flag will return a correct pipeline."""
+    cli_runner.invoke(server.commands, ["viz", "--pipeline", "another"])
+    response = client.get("/api/nodes.json")
+    assert response.status_code == 200
+    data = json.loads(response.data.decode())
+    assert data == {"edges": [], "nodes": [], "tags": []}
+
+
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_pipeline_flag_non_existent(cli_runner):
+    """Test that running viz with `--pipeline` flag but the pipeline does not exist."""
+    result = cli_runner.invoke(server.commands, ["viz", "--pipeline", "nonexistent"])
+    assert "Failed to find the pipeline." in result.output
+
+
+@mock.patch("kedro.__version__", "0.15.0")
+def test_viz_kedro15(cli_runner):
+    """Test that running viz in Kedro 0.15.0."""
+
+    def get_project_context(
+        key: str = "context", **kwargs  # pylint: disable=bad-continuation
+    ):  # pylint: disable=unused-argument
+        mocked_context = mock.Mock()
+        mocked_context.catalog = lambda x: None
+        mocked_context.pipeline = create_pipeline()
+        return {"context": mocked_context}[key]
+
+    mock.patch("kedro_viz.server.get_project_context", new=get_project_context).start()
+    result = cli_runner.invoke(server.commands, "viz")
+    assert result.exit_code == 0, result.output
+
+
+@mock.patch("kedro.__version__", "0.15.0")
+def test_viz_kedro15_pipeline_flag(cli_runner):
+    """Test that running viz with `--pipeline` flag in Kedro 0.15.0."""
+
+    def get_project_context(
+        key: str = "context", **kwargs  # pylint: disable=bad-continuation
+    ):  # pylint: disable=unused-argument
+        return {"context": mock.Mock()}[key]
+
+    mock.patch("kedro_viz.server.get_project_context", new=get_project_context).start()
+    result = cli_runner.invoke(server.commands, ["viz", "--pipeline", "another"])
+    assert "`--pipeline` flag was provided" in result.output
+
+
+@mock.patch("kedro.__version__", "0.15.0")
+def test_viz_kedro15_invalid(cli_runner):
+    """Test that running viz in Kedro 0.15.0,
+    and it is outside of a Kedro project root."""
+
+    def get_project_context(key: str, **kwargs):
+        raise KedroContextError
+
+    mock.patch("kedro_viz.server.get_project_context", new=get_project_context).start()
+    result = cli_runner.invoke(server.commands, "viz")
+    assert "Could not find a Kedro project root." in result.output
```

Comment:
Just realised we're using `unittest` for mocking instead of `pytest`. Could we use `mocker.patch` instead of mock to make it consistent?
We start a patch but never stop it, so I assume it's fine to just do mocker.patch("this")

Reply:
As discussed offline, I will address mock -> mocker stuff in a separate PR :)